rfcs/text/0032-structured-benchmark-programs.md

```diff
@@ -0,0 +1,224 @@
+- Comparisons are not simple: For users, it might not be stratighforward to know what to compare against when compiling these structured programs. For a full, fair evaluation, a meaningful baseline needs to be established and a more precise methodology for comparison is likely necessary.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+Potential alternatives include:
+- *Implementing structured benchmarks using OpenQASM 3*: While it is possible to represent all desired structured program features in OpenQASM 3, many of these features are only optional and not widely supported by existing backends. Furthermore, using such a representation runs the risk of larger specification changes in the future that could break compatibility.
+- *Using more involved quantum programming frameworks like PennyLane*: While frameworks like PennyLane can express structured quantum programs, they often come with additional abstractions and dependencies that are not easily translatable to arbitrary other representations. As the goal of Jeff is to be a widely compatible exchange format, it mitigates this risk by allowing easier compatibility with different representations.
+- *Not implementing structured benchmarks at all*: Sooner or later, the quantum computing community will need to address the challenges of structured control flow in quantum programs. By not providing a benchmark suite, we risk slowing down the progress in this area and missing out on opportunities to drive compiler development.
+
+As the reasoning above shows, the proposed design of structured benchmark programs in Jeff is the most effective way to address the need for evaluating and improving quantum compilers' support for advanced control flow features.
+
+# Prior art
+[prior-art]: #prior-art
+
+A variety of benchmark suites exist already in the quantum computing community, such as:
+- [UCC Bench](https://github.com/unitaryfoundation/ucc-bench): A command-line utility designed to benchmark and compare the performance of various quantum compilers, with a particular focus on the ucc compiler.
+- [MQT Bench](https://github.com/munich-quantum-toolkit/bench): A collection of benchmark circuits in qiskit format to evaluate quantum software tools.
+- [metriq-gym](https://github.com/unitaryfoundation/metriq-gym): A framework for implementing and running standard quantum benchmarks.
+
+While many of these benchmark suites are widely used and provide valuable insights into quantum compiler performance, they often lack support for structured control flow features such as dynamic loops and conditionals.
+
+To the best of our knowledge, there is currently no widely adopted benchmark suite that specifically targets the evaluation of quantum compilers' support for structured control flow operations, although the topic has been discussed in various community gatherings and conferences.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
```

Comment:
Process question -- are these blocking and need to be resolved prior to accepting the RFC and starting work?